chips/litex_vexriscv/src/chip.rs

```diff
@@ -0,0 +1,223 @@
+//! High-level setup and interrupt mapping for the chip.
+
+use core::fmt::Write;
+use kernel;
+use kernel::common::registers::FieldValue;
+use kernel::debug;
+use kernel::hil::time::Alarm;
+use kernel::InterruptService;
+use rv32i::csr::{mcause, mie::mie, mip::mip, CSR};
+use rv32i::syscall::SysCall;
+use rv32i::PMPConfigMacro;
+
+// TODO: Rename this module? Is this really a PLIC?
+use crate::plic;
+
+// TODO: Actually implement the PMP
+PMPConfigMacro!(4);
+
+pub struct LiteXVexRiscv<A: 'static + Alarm<'static>, I: 'static + InterruptService<()>> {
+    soc_identifier: &'static str,
+    userspace_kernel_boundary: SysCall,
+    _pmp: PMP,
+    scheduler_timer: kernel::VirtualSchedulerTimer<A>,
+    interrupt_service: &'static I,
+}
+
+impl<A: 'static + Alarm<'static>, I: 'static + InterruptService<()>> LiteXVexRiscv<A, I> {
+    pub unsafe fn new(
+        soc_identifier: &'static str,
+        alarm: &'static A,
+        interrupt_service: &'static I,
+    ) -> Self {
+        Self {
+            soc_identifier,
+            userspace_kernel_boundary: SysCall::new(),
+            _pmp: PMP::new(),
+            scheduler_timer: kernel::VirtualSchedulerTimer::new(alarm),
+            interrupt_service,
+        }
+    }
+
+    pub unsafe fn enable_plic_interrupts(&self) {
+        plic::enable_interrupts();
```

Comment:
The interrupt controller should be included in the `Chip` (see the RISC-V boards today). Also it doesn't sound like a PLIC.